src/components/expanded-state/chart/ChartExpandedStateHeader.js

```diff
@@ -0,0 +1,137 @@
+import lang from 'i18n-js';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import styled from 'styled-components/primitives';
+import EditOptions from '../../../helpers/editOptionTypes';
+import { toFixedDecimals } from '../../../helpers/utilities';
+import { useAccountSettings, useCoinListEditOptions } from '../../../hooks';
+import { colors, padding } from '../../../styles';
+import { magicMemo } from '../../../utils';
+import { CoinIcon } from '../../coin-icon';
+import { ContextCircleButton } from '../../context-menu';
+import { Icon } from '../../icons';
+import { ColumnWithMargins, Row, RowWithMargins } from '../../layout';
+import { TruncatedText } from '../../text';
+
+const noPriceData = 'No price data';
+
+const Container = styled(ColumnWithMargins).attrs({
+  margin: 12,
+})`
+  ${padding(0, 19, 30)};
+`;
+
+const Title = styled(TruncatedText).attrs(({ color = colors.dark }) => ({
+  color,
+  letterSpacing: 'roundedTight',
+  size: 'big',
+  weight: 'bold',
+}))``;
+
+const Subtitle = styled(TruncatedText).attrs(
+  ({ color = colors.alpha(colors.blueGreyDark, 0.8) }) => ({
+    color,
+    letterSpacing: 'roundedTight',
+    size: 'larger',
+    weight: 'medium',
+  })
+)``;
+
+const ChartExpandedStateHeader = ({
+  address,
+  change,
+  chartPrice,
+  color = colors.dark,
+  isPositiveChange,
+  latestPrice = noPriceData,
+  name,
+  shadowColor,
+  symbol,
+  uniqueId,
+}) => {
+  const { nativeCurrencySymbol } = useAccountSettings();
+  const {
+    currentAction,
+    pushSelectedCoin,
+    setHiddenCoins,
+    setPinnedCoins,
+  } = useCoinListEditOptions();
+
+  useEffect(() => {
+    pushSelectedCoin(uniqueId);
+  }, [currentAction, pushSelectedCoin, uniqueId]);
+
+  const coinIconShadow = useMemo(
+    () => [[0, 4, 12, shadowColor || color, 0.3]],
+    [color, shadowColor]
+  );
+
+  const contextButtonOptions = useMemo(
+    () => [
+      `📌️ ${currentAction === EditOptions.unpin ? 'Unpin' : 'Pin'}`,
+      `🙈️ ${currentAction === EditOptions.unhide ? 'Unhide' : 'Hide'}`,
+      lang.t('wallet.action.cancel'),
+    ],
+    [currentAction]
+  );
+
+  const formattedChange = useMemo(
+    () => `${Math.abs(Number(toFixedDecimals(change, 2)))}%`,
+    [change]
+  );
+
+  const formattedPrice = useMemo(() => {
+    const decimals = ((latestPrice || '').split('.')[1] || []).length;
```

Comment:
if for some reason latestPrice is a number instead of a string (for ex. `2`) it  will throw an exception.
Maybe be worth casting it unless you think it's safe.

Reply:
btw latestPrice is coming from asset?.native?.price.display which will always be a string